### PR TITLE
Enable host networking for provisioner

### LINKED ISSUE #3167: Helm option to to enable host networking for provisioner (hostNetwork: true)
# Describe the feature you'd like to have #

Currently the ceph-csi-rbd-provisioner is running on a pod-network which is problematic if the CEPH-Network is not reachable from the pod-network, but only from the host-network. While this might not be an issue with solutions like rook, it is a problem in our deployment (historically our podNetwork is IPv4 only, while Ceph is IPv6, and migrating to a Dual-Stack pod-network is no option yet). The ceph-csi-rbd-nodeplugin has hostNetwork: true, so this is running fine.

# What is the value to the end user? (why is it a priority?) #
Support scenarios where for some reasons the pod-network can't ro should not be able to communicate with ceph.

# Additional context #
Currently we are using a customized version of the helm-chart, which is a little bit cumbersome.

### PR DESCRIPTION
**Pull request by iceman91176**
# Describe what this PR does #
Adds the possibility in the helm-chart to enable hostNetworking. Useful for cases where the podNetwork has no access to ceph, but the host has.

## Is there anything that requires special attention ##
No ;-) 

## Related issues ##

Fixes: #3167 

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

### CODE REVIEW

**Comment by yati1998** [comment]
@iceman91176 please check the [Commit guidelines](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages) to pass the CI.

File: charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml [hunk 1]
```diff
@@ -48,6 +48,9 @@ spec:
               topologyKey: "kubernetes.io/hostname"
 {{- end }}
       serviceAccountName: {{ include "ceph-csi-cephfs.serviceAccountName.provisioner" . }}
+{{- if .Values.provisioner.enableHostNetwork }}
+      hostNetwork: true
+{{- end }}
```
Comment (Madhu-1) [inline charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml]:
```suggestion
hostNetwork:  {{ .Values.provisioner.enableHostNetwork }}
```

File: charts/ceph-csi-rbd/values.yaml [hunk 2]
```diff
@@ -164,6 +164,10 @@ provisioner:
   # system-cluster-critical which is less priority than system-node-critical
   priorityClassName: system-cluster-critical
 
+  # enable hostnetwork for provisioner pod. default is false
+  # useful for deployments where the podNetwork has no access to ceph
+  enableHostNetwork: false
```
Comment (Madhu-1) [inline charts/ceph-csi-rbd/values.yaml:169]: add documentation for this in https://github.com/ceph/ceph-csi/blob/devel/charts/ceph-csi-rbd/README.md and  https://github.com/ceph/ceph-csi/blob/devel/charts/ceph-csi-cephfs/README.md

**Review by Madhu-1 — CHANGES_REQUESTED** [review]
@iceman91176 Thanks for the contribution, can you please remove extra commits which is not related to this one.

File: charts/ceph-csi-rbd/values.yaml (hunk 2 above)
Comment (iceman91176) [reply charts/ceph-csi-rbd/values.yaml:169]: Hm, i don't know where that additional commit came from. I'll take a look at it.

File: charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml (hunk 1 above)
Comment (iceman91176) [reply charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml]: Did as suggested. Way simpler..

File: charts/ceph-csi-rbd/values.yaml (hunk 2 above)
Comment (iceman91176) [reply charts/ceph-csi-rbd/values.yaml:169]: Did as suggested. Way simpler..

**Review by Madhu-1 — COMMENTED** [review]
changes LGTM. can you please squash the commits into 1? Sadly we don't have CI to test helm upgrades, did you get a chance to verify the helm upgrade (cephcsi) from the older version to a version for this change?

**Comment by iceman91176** [comment]
Well, i performed a dry-run for install and upgrade . That looked good to me.
Commits have been squashed

**Review by Madhu-1 — COMMENTED** [review]
You might need to update commit message as below

```
helm: enable host networking for provisioner

Adds the possibility in the helm-chart to enable hostNetworking
for provider pods.

Signed-off-by: Carsten Buchberger <c.buchberger@witcom.de>
```

**Review by nixpanic — APPROVED** [review]

**Review by Madhu-1 — APPROVED** [review]

**Review by yati1998 — APPROVED** [review]

**Comment by Rakshith-R** [comment]
@mergifyio rebase

**Pull request merged**